### ISSUE TITLE
Should warn about risky `curl` invocations

### ISSUE THREAD

**Issue by walles** [koalaman/shellcheck#2100]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

# Bad
curl "https://www.kth.se/doesntexist"

# Good
curl --fail --max-time 15 "https://www.kth.se/doesntexist"
```

#### Here's what shellcheck currently says:
```sh
$ shellcheck myscript
No issues detected!

$
```

#### Here's what I wanted or expected to see:
* Using `curl` without `--fail` will silently miss error HTTP code responses. So your script will just keep executing even if the page was not found.
* Using `curl` without `--max-time SECONDS` or `-m SECONDS` can hang forever.
